src/overlays/actors/ovl_Obj_Takaraya_Wall/z_obj_takaraya_wall.c

```diff
@@ -30,39 +50,444 @@ ActorInit Obj_Takaraya_Wall_InitVars = {
     (ActorFunc)ObjTakarayaWall_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80ADA2A0 = {
-    { COLTYPE_NONE, AT_NONE, AC_NONE, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_2, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_NONE, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 40, 120, 0, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80ADA2CC[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+#define TAKARAYA_WALL_ROWS 11
+#define TAKARAYA_WALL_COLUMNS 8
+
+typedef struct ObjTakarayaWallSpace {
+    /* 0 */ u8 row;
+    /* 1 */ u8 column;
+} ObjTakarayaWallSpace; // size = 0x2
+
+typedef enum ObjTakarayaWallSpaceState {
+    /* 0 */ TAKARAYA_WALL_INACTIVE,
+    /* 1 */ TAKARAYA_WALL_RISING,
+    /* 2 */ TAKARAYA_WALL_FALLING
+} ObjTakarayaWallSpaceState;
+
+s32 sPathBuilderIndex;
+ObjTakarayaWallSpace sPathBuilder[TAKARAYA_WALL_ROWS * TAKARAYA_WALL_COLUMNS];
+s32 D_80ADA508; // Set but never used
+s32 sPathBuilderReachedStart;
+Vec3f sAudioPositions[TAKARAYA_WALL_ROWS][TAKARAYA_WALL_COLUMNS];
+u8 sSpaceStates[TAKARAYA_WALL_ROWS][TAKARAYA_WALL_COLUMNS];
+f32 sWallHeights[TAKARAYA_WALL_ROWS][TAKARAYA_WALL_COLUMNS]; // -10.0f means the wall does not raise (the space is open)
```

Comment:
Since these aren't static, could you add the file name prefix to the variables? The names seem to generic